Replace existing codebase with new-log-viewer and update docs. (#101)
y-scope/yscope-log-viewer: README.md

Author: I was thinking clp-ffi-go is not a logging library so i intentionally left it. Do we want to add clp-ffi-go?

Author: right, not sure how i was thinking about syntax highlighting when i first looked at that lol

Author: https://github.com/y-scope/yscope-log-viewer/issues/102

Author: Discussed offline that we might want to keep the line

Author: I get what you mean now. Let's move some of those into dev-guide.

logs with syntax highlighting, and generate direct links to specific log events.

See the [features in development](#features-in-development) section for upcoming
features.

# Online Demo

* A demo of the log viewer can be found at https://yscope.com/log-viewer
* The demo loads a Hadoop YARN log file from the [hive-24hrs] log dataset.
  * More info on the dataset and other datasets can be found [here][datasets].
* To open an IR stream, drag and drop it onto the log viewer or use the open
  file dialog.

# Generating IR Stream Logs

IR stream log files can currently be generated using these libraries:

* [Log4j Logging Library][log4j1-appenders]
* [Logback Logging Library][logback-appenders]
* [Python Logging Library][clp-loglib-py]
* Golang Logging Library (in development)

# How does it work?

The log viewer is written using the ReactJS framework and uses the open source
[monaco-editor] and [clp-ffi-js].

The viewer spawns a worker to assist with computationally intensive tasks such
as:

* Deserializing the file and creating an index of logs events.
* Paginating the indexed logs.
* Decoding the deserialized log events into plain text.

> Tasks are passed to the worker as needed and changes are rendered in the UI.

Author: To me the suggested text seems a bit less precise

# YScope Log Viewer

![Open bug reports][bugs-shield]
![Open feature requests][feature-requests-shield]
![CLP on Zulip][zulip-shield]

`yscope-log-viewer` is a tool that can be used to view log files. It currently
supports [CLP][clp-repo]'s compressed log files (IR streams) and JSON log files.
The viewer can be used to navigate the log file, filter by log level, view the
logs with syntax highlighting, and generate direct links to specific log events.

See the [features in development](#features-in-development) section for upcoming
features.

# Online Demo

* A demo of the log viewer can be found at https://yscope.com/log-viewer
* The demo loads a Hadoop YARN log file from the [hive-24hrs] log dataset.

>     * More info on the dataset and other datasets can be found [here][datasets].

Reviewer: ```suggestion
  * More info on the dataset and other datasets can be found [here][datasets].
```

* To open an IR stream, drag and drop it onto the log viewer or use the open
  file dialog.

# Generating IR Stream Logs

IR stream log files can currently be generated using these libraries:

* [Log4j Logging Library][log4j1-appenders]
* [Logback Logging Library][logback-appenders]
* [Python Logging Library][clp-loglib-py]
* Golang Logging Library (in development)

# How does it work?

The log viewer is written using the ReactJS framework and uses the open source
[monaco-editor] and [clp-ffi-js].

The viewer spawns a worker to assist with computationally intensive tasks such
as:

* Deserializing the file and creating an index of logs events.
* Paginating the indexed logs.
* Decoding the deserialized log events into plain text.

Tasks are passed to the worker as needed and changes are rendered in the UI.

# Providing Feedback

You can use GitHub issues to [report a bug][report-bug] or
[request a feature][request-feature].

Join us on [Zulip][zulip] to chat with developers and other community members.

# Developer Guide

You can find our docs source in `docs/dev-guide`.

# Features in Development

* Pretty printing to enhance readability of structured data in the logs.

> * Dashboard to visualize distribution of log types and log levels.

Reviewer: ```suggestion
* A dashboard to visualize distribution of log types and log levels.
```

> * Opening log files which are larger than 2 GB when uncompressed.

Author: ```suggestion
```

* Support for plain text, archived log files, and other requested formats.

> * Search across file, multiple files or within provided time ranges.

Reviewer: ```suggestion
* Searching within a file, multiple files, or within provided time ranges.
```

> * Infinite scrolling instead of using pagination.

Reviewer: ```suggestion
* Infinite scrolling instead of pagination.
```

# YScope Log Viewer

![Open bug reports][bugs-shield]
![Open feature requests][feature-requests-shield]
![CLP on Zulip][zulip-shield]

`yscope-log-viewer` is a tool that can be used to view log files. It currently
supports [CLP][clp-repo]'s compressed log files (IR streams) and JSON log files.
The viewer can be used to navigate the log file, filter by log level, view the
logs with syntax highlighting, and generate direct links to specific log events.

> To start a local server for accessing the application locally, refer to
> `docs/dev-guide/building.md`.

Reviewer: ```suggestion
To start a local server for accessing the application locally, refer to our
[build guide](docs/dev-guide/building.md).
```

* The demo loads a Hadoop YARN log file from the [hive-24hrs] log dataset.
  * More info on the dataset and other datasets can be found [here][datasets].
* To open an IR stream, drag and drop it onto the log viewer or use the open
  file dialog.

# Generating IR Stream Logs

IR stream log files can currently be generated using these libraries:

* [Log4j Logging Library][log4j1-appenders]
* [Logback Logging Library][logback-appenders]
* [Python Logging Library][clp-loglib-py]
* Golang Logging Library (in development)

# How does it work?

The log viewer is written using the ReactJS framework and uses the open source
[monaco-editor] and [clp-ffi-js].

The viewer spawns a worker to assist with computationally intensive tasks such
as:

* Deserializing the file and creating an index of logs events.
* Paginating the indexed logs.
* Decoding the deserialized log events into plain text.

Tasks are passed to the worker as needed and changes are rendered in the UI.

# Providing Feedback

You can use GitHub issues to [report a bug][report-bug] or
[request a feature][request-feature].

Join us on [Zulip][zulip] to chat with developers and other community members.

# Contributing

> You can find our developer guide in `docs/dev-guide`.

Reviewer: ```suggestion
See the docs in our [developer guide](docs/dev-guide).
```

# Features in Development

> * Pretty printing to enhance readability of structured data in the logs.

Reviewer: ```suggestion
* Pretty printing to enhance the readability of structured data in the logs.
```

> * A dashboard to visualize distribution of log types and log levels.

Reviewer: ```suggestion
* A dashboard to visualize the distribution of log types and log levels.
```

# How does it work?

`yscope-log-viewer` is written using the ReactJS framework and uses the open 
source [monaco-editor](https://github.com/microsoft/monaco-editor). For decoding
files in CLP formats, APIs provided by
[clp-ffi-js](https://github.com/y-scope/clp-ffi-js) are used.

> To open log files, the viewer spawns a worker to perform the following tasks:

Author: Sure. Let's see if moving it between `Generating IR Stream Logs` and `Providing Feedback` would make more sense.
